cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -431,6 +460,10 @@ async function unusedValidationOptions(ownedCertARN, loadBalancerDNS) {
     // Look for validation options that will be no longer needed by this service.
     const certificates = await serviceCertificates();
     const { certOwned: certPendingDeletion, otherCerts: certInUse } = categorizeCertificates(certificates, ownedCertARN);
+    if (!certPendingDeletion) {
+        // Cannot find the certificate that is pending deletion; perhaps it is deleted already. Exit peacefully.
```

Comment:
If the cert doesn't exist, will it be a problem when trying to delete the cert?

Reply:
```
 await clients.acm().deleteCertificate({ CertificateArn: certARN }).promise().catch(err => {
        if (err.name !== "ResourceNotFoundException") {
            throw err;
        }
    });
```
This error handling should handle it

Reply:
How about the describe logic before trying to delete the cert here? https://github.com/aws/copilot-cli/pull/3132/files#diff-904aa8592d00fc56cc75b3a4e5c3d52404d435129b7717885ece3b8e402b1860R251-R258